### ISSUE TITLE
Storing application state  locally and on server

### ISSUE THREAD

**Issue by kepta** [sen-team8/da-dash#9]
I would like to have da-dash store the session on server.
It should save on both server and locally, and decide which one to use.
We can have it save the entire redux object.
@flibertigibet what do you think?

**Comment by flibertigibet** [comment]
![fb_rearch](https://cloud.githubusercontent.com/assets/13137746/14104023/93b353f4-f5c0-11e5-8dc4-0cbc09a77964.png)

Something like this but firebase instead of mongo. I think it'll be faster and user can work offline

**Comment by kepta** [comment]
Well I don't know what you mean by offline, since it is a web app. Can you explain a bit more.
The diagrams are for a desktop app right?

**Comment by flibertigibet** [comment]
http://thejackalofjavascript.com/re-architecting-a-firebase-app-in-node/

The diagrams are for a web app. 

I was reading from this and what I meant is that is it possible to have some of the user data stored locally on the client's filesystem so that he can use some functionality of the dashboard even when offline. Like his todo list. And later sync the local and server when he's online. 

P.S. I have very little idea about this.

**Comment by kepta** [comment]
Yes exactly, we will store the data at both client side and on server side. The server side copy would be  a clone of the client side.
We would need to implement a way to intelligently manage these two.
